### PR TITLE
English to Hindi: Referencing values with refs

### PR DESCRIPTION
**Pull request by khizarq313**

### CODE REVIEW

File: beta/src/pages/learn/referencing-values-with-refs.md [hunk 1]
```diff
@@ -1,37 +1,37 @@
 ---
-title: 'Referencing Values with Refs'
+title: 'Refs के ज़रिए वैल्यू का रेफरन्स लेना'
 ---
 
 <Intro>
 
-When you want a component to "remember" some information, but you don't want that information to [trigger new renders](/learn/render-and-commit), you can use a *ref*.
+जब आप किसी कंपोनेंट को कुछ जानकारी "याद" करवाना चाहते हों, लेकिन आप नहीं चाहते कि वह जानकारी [नए रेंडर को ट्रिगर करे](/learn/render-and-commit),तब आप एक *ref * का उपयोग कर सकते हैं।
```
Comment (arshadkazmi42) [inline beta/src/pages/learn/referencing-values-with-refs.md:7]:
```suggestion
जब आप किसी कौम्पोनॅन्ट को कुछ जानकारी "याद" रखवाना चाहते हों, लेकिन आप नहीं चाहते कि वह जानकारी [नए रेंडर को ट्रिगर करे](/learn/render-and-commit), तब आप एक *ref * का उपयोग कर सकते हैं।
```

File: beta/src/pages/learn/referencing-values-with-refs.md [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
 
 </Intro>
 
 <YouWillLearn>
 
-- How to add a ref to your component
-- How to update a ref's value
-- How refs are different from state
-- How to use refs safely
+- अपने कंपोनेंट में refs कैसे जोड़ें
```
Comment (arshadkazmi42) [inline beta/src/pages/learn/referencing-values-with-refs.md:13]: `कंपोनेंट` => `कौम्पोनॅन्ट`

update all the references of this in the whole file. You can do a find and replace

Comment (arshadkazmi42) [inline beta/src/pages/learn/referencing-values-with-refs.md:13]:
```suggestion
- अपने कौम्पोनॅन्ट में refs कैसे ऐड करें
```

File: beta/src/pages/learn/referencing-values-with-refs.md [hunk 3]
```diff
@@ -1,37 +1,37 @@
...
+- refs के वैल्यू को कैसे अपडेट करें
```
Comment (arshadkazmi42) [inline beta/src/pages/learn/referencing-values-with-refs.md:14]:
```suggestion
- ref के वैल्यू को कैसे अपडेट करें
```

File: beta/src/pages/learn/referencing-values-with-refs.md [hunk 4]
```diff
@@ -1,37 +1,37 @@
...
+- refs state से कैसे अलग होते हैं
```
Comment (arshadkazmi42) [inline beta/src/pages/learn/referencing-values-with-refs.md:15]:
```suggestion
- ref state से कैसे अलग होते हैं
```

File: beta/src/pages/learn/referencing-values-with-refs.md [hunk 5]
```diff
@@ -1,37 +1,37 @@
...
+- refs को सुरक्षित तरीके से कैसे उपयोग करें
```
Comment (arshadkazmi42) [inline beta/src/pages/learn/referencing-values-with-refs.md:16]:
```suggestion
- ref को सुरक्षित तरीके से कैसे उपयोग करें
```

File: beta/src/pages/learn/referencing-values-with-refs.md [hunk 6]
```diff
@@ -1,37 +1,37 @@
...
 
 </YouWillLearn>
 
-## Adding a ref to your component {/*adding-a-ref-to-your-component*/}
+## अपने कंपोनेंट में refs जोड़ना {/*adding-a-ref-to-your-component*/}
```
Comment (arshadkazmi42) [inline beta/src/pages/learn/referencing-values-with-refs.md:20]:
```suggestion
## अपने कौम्पोनॅन्ट में ref ऐड करना {/*adding-a-ref-to-your-component*/}
```

File: beta/src/pages/learn/referencing-values-with-refs.md [hunk 7]
```diff
@@ -1,37 +1,37 @@
...
 
-You can add a ref to your component by importing the `useRef` Hook from React:
+आप React से `useRef` हुक इम्पोर्ट करके अपने कंपोनेंट में एक refs जोड़ सकते हैं:
```
Comment (arshadkazmi42) [inline beta/src/pages/learn/referencing-values-with-refs.md:22]:
```suggestion
आप React से `useRef` हुक इम्पोर्ट करके अपने कौम्पोनॅन्ट में एक ref ऐड कर सकते हैं:
```

File: beta/src/pages/learn/referencing-values-with-refs.md [hunk 8]
```diff
@@ -1,37 +1,37 @@
...
 
 ```js
 import { useRef } from 'react';
 ```
 
-Inside your component, call the `useRef` Hook and pass the initial value that you want to reference as the only argument. For example, here is a ref to the value `0`:
+अपने कंपोनेंट के अंदर, `useRef` हुक को कॉल करें और प्रारंभिक वैल्यू पास करें जिसे आप केवल तर्क के रूप में रेफरन्स करना चाहते हैं। उदाहरण के लिए, यहां वैल्यू `0` के लिए एक refs है:
```
Comment (arshadkazmi42) [inline beta/src/pages/learn/referencing-values-with-refs.md:28]:
```suggestion
अपने कौम्पोनॅन्ट के अंदर, `useRef` हुक को कॉल करें और इनिशियल वैल्यू पास करें जिसे आप केवल आर्गुमेंट के रूप में रेफरन्स करना चाहते हैं। उदाहरण के लिए, यहां वैल्यू `0` के लिए एक ref है:
```

File: beta/src/pages/learn/referencing-values-with-refs.md [hunk 9]
```diff
@@ -1,37 +1,37 @@
...
 
 ```js
 const ref = useRef(0);
 ```
 
-`useRef` returns an object like this:
+`useRef` एक इस तरह के ऑब्जेक्ट को वापस करता है:
```
Comment (arshadkazmi42) [inline beta/src/pages/learn/referencing-values-with-refs.md:34]:
```suggestion
`useRef` एक इस तरह के ऑब्जेक्ट को रिटर्न करता है:
```

**Review by arshadkazmi42 — CHANGES_REQUESTED** [review]
Thank you for working on this. I have added review comments till line 44. 

Let's get these fixed then we can continue further with the review.

To know more about our review process check #23

**Comment by khizarq313** [comment]
> @khizarq313 I have added some review comments. Can you fix those, also can you fix the conflict we are seeing in one of the file.

sure, I will take a look at it asap.

**Comment by arshadkazmi42** [comment]
@khizarq313 Looks like you did a force push which removed all the changes and I am seeing the PR is closed. Are you going to open a new PR?

**Comment by khizarq313** [comment]
i was trying to fix the conflict and there was a massage saying that my repository is one commit ahead, i thought that might be causing the conflict so i pressed the sync button...and this happened.. sorry for the trouble i caused...
I am working on translation and yes i will open new pr.

**Comment by khizarq313** [comment]
I am correcting my translation,also i am reffering hindi glossary page 
(https://github.com/reactjs/hi.react.dev/issues/2),

i still don't know the reason behind the previously faced conflict,
I need you to clarify me somethings,
this is my first contribution,
i fork the hi.reactjs.org repo then clone it locally,
After translation i should commit it and ask for review through pull request?
am i doing it correctly, or should i need to correct myself ?
any clarification is appricate and i am sorry for my inconvinence.

**Comment by arshadkazmi42** [comment]
Yes. it is correct. Only one thing you need to consider is your fork could get out of sync from the main repository and you will have to sync it with the main repository 

You can refer this stackoverflow on how to do this https://stackoverflow.com/a/7244456/3136282

Or I have created a cli tool for this [upfork](https://github.com/gh-conf/upfork) you could also use this.

**Comment by khizarq313** [comment]
sure, I will go with cli tool as it seems simple and easy to use.
Thank you!

**Comment by khizarq313** [comment]
Hey!
What should i write in hindi for word "unreliable",

English:
This is why reading `ref.current` during render leads to unreliable code. If you need that, use state instead.
Hindi:
इसीलिए रेंडर के दौरान `ref.current` को पढ़ने से कोड अविश्वसनीय (unreliable) होजाता है। अगर आपको उसकी जरूरत है, तो ref के बजाय state का इस्तेमाल करें।

one more thing...
how should i write this: "escape hatch" or "एस्केप हैच"

**Comment by khizarq313** [comment]
also what should i use for word: synchronously or तुल्यकालिकत ?

**Comment by arshadkazmi42** [comment]
@khizarq313 We can use following

`synchronously` => `सिंक्रोनोसली`
 `unreliable` => `अनरीलाऐबल`

`एस्केप हैच` will be better

**Pull request closed without merging**